src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -176,31 +178,40 @@ private static String anonymousLogin(final Map<String, String> propertiesReadFro
         return "Lobby watcher usernames must end with 'lobby_watcher'";
       }
       final String hostName = userName.substring(0, userName.indexOf(InGameLobbyWatcher.LOBBY_WATCHER_NAME));
-      final String issue = UserDao.validateUserName(hostName);
-      if (issue != null) {
-        return issue;
+
+      if (!DbUser.isValidUserName(hostName)) {
+        return DbUser.getInvalidUserNameReason(hostName);
       }
     } else {
-      final String issue = UserDao.validateUserName(userName);
-      if (issue != null) {
-        return issue;
+      if (DbUser.isValidUserName(userName)) {
+        return null;
+      } else {
+        return DbUser.getInvalidUserNameReason(userName);
       }
     }
     return null;
   }
 
   private static String createUser(final Map<String, String> propertiesReadFromClient, final String userName) {
-    final String email = propertiesReadFromClient.get(EMAIL_KEY);
-    final String hashedPassword = propertiesReadFromClient.get(HASHED_PASSWORD_KEY);
-    final DbUserController controller = new DbUserController();
-    final String error = UserDao.validate(userName, email, hashedPassword);
-    if (error != null) {
-      return error;
+    final DbUser user = new DbUser(
+        new DbUser.UserName(userName),
+        new DbUser.UserEmail(propertiesReadFromClient.get(EMAIL_KEY)));
+
+    if (!user.isValid()) {
+      return user.getValidationErrorMessage();
+    }
+
+    final HashedPassword password = new HashedPassword(propertiesReadFromClient.get(HASHED_PASSWORD_KEY));
+    if (!password.isValidSyntax()) {
+      return "Password is not hashed correctly";
     }
+
     try {
-      controller.createUser(userName, email, hashedPassword, false);
+      new DbUserController().createUser(user, password);
       return null;
     } catch (final IllegalStateException ise) {
+      s_logger.severe("Error creating user: " + user + ", err: " + ise.getMessage());
+      ise.printStackTrace();
```

Comment:
@DanVanAtta As @ron-murhammer pointed out, I forgot to dupe this question in my second round of comments.  While you're focused on this issue at the present, could you please review so it gets in this PR if appropriate?